src/server/scripts/Commands/cs_cast.cpp

```diff
@@ -191,13 +202,7 @@ class cast_commandscript : public CommandScript
         if (!*args)
             return false;
 
-        Unit* target = handler->getSelectedUnit();
-        if (!target)
-        {
-            handler->SendSysMessage(LANG_SELECT_CHAR_OR_CREATURE);
-            handler->SetSentErrorMessage(true);
-            return false;
-        }
+        Unit* player = handler->GetSession()->GetPlayer();
```

Comment:
That makes no sense if it is **self cast**, otherwise it would be **cast (to victim)** or **cast back**...